chain/consensus/mir/mine.go

```diff
@@ -57,17 +79,70 @@ func Mine(ctx context.Context, addr address.Address, api v1api.FullNode) error {
 
 		// Miner (leader) for an epoch is assigned deterministically using round-robin.
 		// All other validators use the same Miner in the block.
-		epochMiner := sm.Validators[int(base.Height())%len(sm.Validators)].Addr
+		epochMiner := m.GetBlockMiner(base.Height())
```

Comment:
Can the epoch miner be established only just before creating the block? That would be much cleaner, as it is only used there, plus it would be more obviously synchronized with the current state of the membership.

Reply:
Fixed - https://github.com/filecoin-project/eudico/pull/234/commits/a014449298f6902fa97a65f764ed6cfed8a45408